src/main/java/com/faforever/client/reporting/ReportingServiceImpl.java

```diff
@@ -14,8 +18,22 @@
 
   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
+  @Override
+  public void setAutoReportingUser(String username, int userId) {
+    Context context = Sentry.getContext();
+    UserBuilder userBuilder = new UserBuilder()
+        .setUsername(username)
+        .setId(String.valueOf(userId));
+    context.setUser(userBuilder.build());
+  }
+
   @Override
   public void reportError(Throwable e) {
-    logger.warn("Reporting has not yet been implemented");
+    Sentry.capture(e);
+    LoggerFactory.getLogger((MethodHandles.lookup().lookupClass())).warn("Logging error to sentry: {}", e.toString());
```

Comment:
If you use `@Slf4J` on the class, then you can call `log.warn()`.

However, this should be logged *before* sentry is called. Otherwise, it should be past tense (but I prefer before because if something goes wrong, we don't lose the initial exception). And logging the stack trace is preferred over logging the exception message.

Reply:
With a limit of 10k exceptions, I do not think that automated reporting is a good idea

Reply:
I disagree, there are a variety of rate limiting options in sentry. Also, if we feel that sentry provides enough value we can easily spin up our own instance. https://hub.docker.com/_/sentry/
As noted in their installation process it can store quite a bit of events in about 1.5 tb 
https://docs.sentry.io/server/installation/
I don't think we'll hit anywhere near that and also this might be a real easy way to allow multi user access.